### PR TITLE
feat: read and write support for forest.car.zst files

### PR DESCRIPTION
**Pull request by aatifsyed**
## Summary of changes

Changes introduced in this pull request:

- Add `snapshot compress` command for generating `.forest.car.zst` files.
- Change all other commands to accept either `.car`, `.car.zst` or `.forest.car.zst`.

## Reference issue to close (if applicable)

Closes

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

**Review by elmattic — APPROVED** [review]
Looks good but it's really big for a single PR!

File: src/db/car/forest.rs [hunk 1]
```diff
@@ -0,0 +1,420 @@
...
+            .decode(&mut header_zstd_frame)?
+            .ok_or(invalid_data("malformed uvibytes"))?;
+        let header = from_slice::<CarHeader>(&block_frame)?;
+
+        let index = CarIndex::open(mk_reader()?, footer.index)?;
+        let inner = ForestCarInner {
+            // new_reader: Box::new(mk_reader),
+            reader,
+            frame_cache: LruCache::new(nonzero!(1024_usize)),
+            write_cache: ahash::HashMap::default(),
```
[the remark this answers is not in the document]
Comment (lemmih) [reply src/db/car/forest.rs:104]: The write cache is only used for testing. To get persistence, the ForestCar store must be used with ParityDB. Then all the writes go to ParityDB, and ForestCar will be read-only.

[the remark this answers is not in the document]
Comment (elmattic) [reply src/db/car/forest.rs:104]: Ah I see, thanks.

File: benches/car-index.rs [hunk 2]
```diff
@@ -0,0 +1,66 @@
+// Copyright 2019-2023 ChainSafe Systems
```
Comment (LesnyRumcajs) [inline benches/car-index.rs:1]: Is this benchmark compiled in the CI?

File: src/utils/db/car_index/car_index_builder.rs [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    table: Vec<Slot>,
+    pub longest_distance: u64,
+    pub collisions: u64,
+    capacity: usize,
+}
+
+impl CarIndexBuilder {
+    // Number of buckets given `len` number of elements
+    pub fn capacity_at(len: usize) -> usize {
+        len * 100 / 81
```
Comment (LesnyRumcajs) [inline src/utils/db/car_index/car_index_builder.rs:18]: It's not clear where are those numbers coming from to me.

**Review by LesnyRumcajs — APPROVED** [review]
I'd have to spend at least a few days to understand this PR, so accepting on the basis of "I didn't see any intentional malicious code here", faith and a green checkmark.

File: benches/car-index.rs (hunk 2 above)
Comment (lemmih) [reply benches/car-index.rs:1]: It is not. :( We'll fix that in a follow-up PR.

File: src/utils/db/car_index/car_index_builder.rs (hunk 3 above)
Comment (lemmih) [reply src/utils/db/car_index/car_index_builder.rs:18]: It's the load-factor. I'll add some more documentation.

**Pull request merged**